aio/content/examples/toh-pt5/src/app/hero.service.ts

```diff
@@ -12,16 +12,21 @@ export class HeroService {
   constructor(private messageService: MessageService) { }
 
   getHeroes(): Observable<Hero[]> {
-    // TODO: send the message _after_ fetching the heroes
+    const heroes = of(HEROES);
     this.messageService.add('HeroService: fetched heroes');
-    return of(HEROES);
+    return heroes;
   }
 
   // #docregion getHero
   getHero(id: number): Observable<Hero> {
-    // TODO: send the message _after_ fetching the hero
+    const hero = HEROES.find(h => h.id === id);
+
+    if (hero === undefined) {
+      throw new Error(`Failed to fetch hero id=${id}`);
+    }
+
```

Comment:
Just curious if we can use a non-null assertion statement to keep things simple in the example code:
```suggestion
    const hero = HEROES.find(hero => hero.id === id)!;
```
If we throw an error in a service, I guess we should also illustrate how to handle that, which goes beyond simple examples.

Reply:
I switched so many times between the two options while working on this locally :grin:
Neither option is ideal for the tutorial (non-null assertion is too magic and covers up cases where the ID does not exist and throwing errors without handling them offers a bad UX).

This code is [enhanced in the next chapter][1] of the tutorial to use `HttpClient` for fetching the hero and handling errors, so I don't think it is worth introducing error handling here. How about adding an `as Hero` assertion and add a `TODO` comment to handle the case where the hero cannot be found?

[1]: https://github.com/angular/angular/blob/7d5b60e4097195168c454c18d1b9352ea8c9b9a4/aio/content/examples/toh-pt6/src/app/hero.service.ts#L39-L49

Reply:
I've pushed a fixup commit. WDYT?

Reply:
> I've pushed a fixup commit. WDYT?

Looks great, just added a note with a proposal on rephrasing comments.